test/unit/ts.mjs

```diff
@@ -0,0 +1,55 @@
+import {testPreProcessorExports, batchTest} from '../test-utils.mjs';
+
+import ts from 'typescript';
+
+async function main() {
+  try {
+    let tsOptionsList = [
+      {
+        compilerOptions: {
+          target: 'esnext',
+          module: 'esnext',
+        },
+      },
+      {
+        compilerOptions: {
+          target: 'esnext',
+          module: 'esnext',
+        },
+      },
+      {
+        compilerOptions: {
+          target: ts.ScriptTarget.ESNext,
+          module: ts.ModuleKind.CommonJS,
+        },
+      },
+      {
+        compilerOptions: {
+          target: ts.ScriptTarget.ESNext,
+          module: ts.ModuleKind.ES2015,
+        },
+      },
+    ];
+
+    let preProcessorTypeScript = await import(
+      '../../examples/loaders/preprocessor-typescript.mjs'
+    );
+
+    await testPreProcessorExports(preProcessorTypeScript, tsOptionsList[0]);
+
+    await batchTest(
+      preProcessorTypeScript,
+      tsOptionsList,
+      [
+        'test/fixtures/tsmodule1.ts',
+        'test/fixtures/tsmodule2.ts',
+        'test/fixtures/tsmodule3.ts',
+        'test/fixtures/tsmodule4.ts',
+      ]
+    );
+  } catch (error) {
+    console.log(error);
+  }
+}
+
+main();
```

Comment:
I'm not super familiar with mjs, but is it possible to top-level await? Might want to do that or something to make sure errors get propagated out in a way that TAP will catch.

Reply:
Yup good point, I'll change this to TAP's own async test call, should allow TAP to catch errors that way

Reply:
Changed this pattern and tested for TAP catching errors, good call 👍